test/AssetShareAppTest.js

```diff
@@ -0,0 +1,124 @@
+            // app.address, // app (where?) - The app that holds the role involved in this permission.
+            // INCREMENT_ROLE, // role (what?) - The particular role that the entity is being assigned to in this permission.
+            // appManager, // manager - Can grant/revoke further permissions for this role.
+            // { from: appManager }
+        // )
+        // await acl.createPermission(
+            // ANY_ADDRESS,
+            // app.address,
+            // DECREMENT_ROLE,
+            // appManager,
+            // { from: appManager }
+        // )
+
+        await app.initialize()
+    })
+    
+    it('Initialize with single owner having all shares', async () => {
+        
+        // Assert owner count.
+        var count = await app.getOwnersCount();
+        assert.equal(count, 1);
+        
+        // Assert that the owner owns all shares.
+        var owner = await app.getOwnerAddressByIndex(0);
+        var ownerShares1 = parseInt(await app.getShares({from: owner})); // Getter option 1.
+        var ownerShares2 = parseInt(await app.getSharesByAddress(owner)); // Getter option 2.
+        assert.equal(ownerShares1, parseInt(await app.TOTAL_SHARES()));
+        assert.equal(ownerShares2, parseInt(await app.TOTAL_SHARES()));
+    });
+    
+    it('Initialize with empty treasury', async () => {
+        var treasuryBalance = parseInt(await app.getTreasuryBalance());
+        assert.equal(treasuryBalance, 0);
+    });
+    
+    it('Add to treasury increases balance', async () => {
+        var treasuryBalanceBefore = parseInt(await app.getTreasuryBalance());
+        await app.treasuryDeposit('InfoMessage', {from: user1, value: 1000});
+        var treasuryBalanceAfter = parseInt(await app.getTreasuryBalance());
+        assert.equal(treasuryBalanceAfter, treasuryBalanceBefore + 1000);
+    });
+    
+    it('Add to treasury increases balance twice', async () => {
+        var treasuryBalanceBefore = parseInt(await app.getTreasuryBalance());
+        await app.treasuryDeposit('InfoMessage', {from: user1, value: 1000});
+        var treasuryBalanceAfter1 = parseInt(await app.getTreasuryBalance());
+        assert.equal(treasuryBalanceAfter1, treasuryBalanceBefore + 1000);
+        await app.treasuryDeposit('InfoMessage', {from: user2, value: 500});
+        var treasuryBalanceAfter2 = parseInt(await app.getTreasuryBalance());
+        assert.equal(treasuryBalanceAfter2, treasuryBalanceBefore + 1000 + 500);
+    });
+    
+    it('Payout to owners', async () => {
+        
+        // Deposit money into the treasury.
+        await app.treasuryDeposit('InfoMessage', {from: user1, value: 1e12});
+        
+        // Payout to the shareholders.
+        var currentTimeSec = Date.now() / 1000;
+        await app.payOwners({timestamp: currentTimeSec + parseInt(await app.payoutPeriod)});
+        
+        // Assert that the owners have received the payout.
+        // TODO - How to get Ether from address?
```

Comment:
I've obtained some information and sent that to @preppas to work on this last week. It can be merged for now, and implemented later. The main purpose of this PR is to add an executable test code framework to the project.